src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -329,6 +333,11 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 	}
 
 	private ExpressionSyntax Transform( InvocationExpressionSyntax invocationExpr) {
+		ITypeSymbol? returnTypeInfo = Model.GetTypeInfo( invocationExpr ).Type;
+		if( returnTypeInfo?.MetadataName == "IAsyncEnumerable`1" && returnTypeInfo.ContainingNamespace.ToString() == "System.Collections.Generic" ) {
+			return invocationExpr.WithExpression( SyntaxFactory.ParseExpression( invocationExpr.Expression.ToString().Replace( "Async", "" ) ) );

```

Comment:
I'm a little nervous about calling `ParseExpression` here... there are unimportant amusing consequences like how `var x = SomethingAsyncSomething() /* blah Async blah */;` would get transformed into `var x = SomethingSomething() /* blah blah */;` I think? (like substitution within the comment)

I wonder if we should just have a state variable on the transformer that tells `RemoveAsyncSuffix` to not just do suffixes but do `.Replace` instead. Then the inside of this if would just set that to true, and we would restore the previous state of that variable on the way out of this function... does that make sense?